extra_bindings.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package charm
+
+import (
+	"fmt"
+
+	"github.com/juju/schema"
+	"github.com/juju/utils/set"
+)
+
+// ExtraBinding represents a bindable endpoint which is not also a relation.
+type ExtraBinding struct {
+	Name string `bson:"name" json:"Name"`
+}
+
+// When specified, the "extra-bindings" section in the metadata.yaml
+// should have the following format:
+//
+// extra-bindings:
+//     "<endpoint-name>":
+//     ...
+// Endpoint names are strings and must not match an existing relation names
+// from the Provides, Requires, or Peers metadata sections. The values beside
+// each endpoint name must be left out (i.e. "foo": <anything> is invalid).
+var extraBindingsSchema = schema.Map(schema.NonEmptyString("binding name"), schema.Nil(""))
+
+func parseMetaExtraBindings(data interface{}) (map[string]ExtraBinding, error) {
+	if data == nil {
+		return nil, nil
+	}
+
+	bindingsMap := data.(map[interface{}]interface{})
+	result := make(map[string]ExtraBinding)
+	for name, _ := range bindingsMap {
+		stringName := name.(string)
```

Comment:
Yes, because parseMetaExtraBindings (like other similar helpers for resources, payloads, storage, etc.) are called once the top-level charm schema has been Coerce()d.